tests/PHPUnit/RevisionStrikeTest.php

```diff
@@ -282,29 +344,282 @@ public function test_get_revision_ids_with_multiple_post_types() {
+
+		// revision IDs for the post ID 1
+		M::wpFunction( 'wp_list_pluck', array(
+			'args' => array(
+				$scrubbed_posts_list["1"],
+				'revision_id',
+				),
+			'times' => 1,
+			'return' => array( 12 ), // delete only one oldest revision, other two are not deleted
+			)
+		);
+
+		// revision IDs for the post ID 2
+		M::wpFunction( 'wp_list_pluck', array(
+			'args' => array(
+				$scrubbed_posts_list["2"],
+				'revision_id',
+				),
+			'times' => 1,
+			'return' => array(  ), // no revisions to delete
+			)
+		);
+
+		// revisions for the post ID 3
+		M::wpFunction( 'wp_list_pluck', array(
+			'args' => array(
+				$scrubbed_posts_list["3"],
+				'revision_id',
+				),
+			'times' => 1,
+			'return' => array( 15, 16 ), // two oldest revisions
+			)
+		);
+
+
+		M::wpPassthruFunction( 'absint' );
+
+		$result = $method->invoke( $instance, 90, 25, 'post', 2 );
+		$wpdb   = null;
+
+		$this->assertEquals( array( 12, 15, 16 ), $result );
+
+		$stats = $property->getValue( $instance );
+		$this->assertEquals( 3, $stats['count'], 'The "count" statistic is not being updated' );
 	}
 
 	public function test_get_revision_ids_returns_early_with_empty_post_types() {
 		$instance = new RevisionStrike;
 		$method   = new ReflectionMethod( $instance, 'get_revision_ids' );
 		$method->setAccessible( true );
 
-		$this->assertEquals( array(), $method->invoke( $instance, 30, 50, '' ) );
+		$this->assertEquals( array(), $method->invoke( $instance, 30, 50, '', 0 ) );
+	}
+
+	/**
+	 * Creates array of mock query results from the wpdb SELECT query
+	 *
+	 * @return array
+	 */
+	private function mock_query_post_and_revision_ids_results() {
+		/*
+			get_results is running:
+			SELECT r.ID as revision_id, r.post_date as revision_date, p.ID as post_id
+		 */
+		$posts_and_revision_objects = array();
+
+		// post id 1 with three revisions
+		$posts_and_revision_objects[] = $this->mock_query_result( 10, '2016-04-01', 1 );
+		$posts_and_revision_objects[] = $this->mock_query_result( 11, '2016-04-10', 1 );
+		$posts_and_revision_objects[] = $this->mock_query_result( 12, '2016-03-01', 1 );
+
+		// post id 2 with two revisions
+		$posts_and_revision_objects[] = $this->mock_query_result( 13, '2016-01-01', 2 );
+		$posts_and_revision_objects[] = $this->mock_query_result( 14, '2016-02-10', 2 );
+
+		// post id 3 with four revisions
+		$posts_and_revision_objects[] = $this->mock_query_result( 15, '2016-03-01', 3 );
+		$posts_and_revision_objects[] = $this->mock_query_result( 16, '2016-02-10', 3 );
+		$posts_and_revision_objects[] = $this->mock_query_result( 17, '2016-05-01', 3 );
+		$posts_and_revision_objects[] = $this->mock_query_result( 18, '2016-05-07', 3 );
+
+		return $posts_and_revision_objects;
+	}
+
+	/**
+	 * Creates mock stdClass based on the SELECT query
+	 *
+	 * @return object
+	 */
+	function mock_query_result( $revision_id, $revision_date, $post_id ) {
```

Comment:
Please add this method's access modifier, since public methods are treated as test cases.